addons/binding/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,263 @@
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.pixometer.internal.config.ReadingInstance;
+import org.openhab.binding.pixometer.internal.serializer.CustomReadingInstanceDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link MeterHandler} is responsible for handling data and measurements of a meter thing
+ *
+ * @author Jerome Luckenbach - Initial contribution
+ */
+public class MeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeterHandler.class);
+
+    private static final String API_BASE_URL = "https://pixometer.io/api/";
+    private static final String API_VERSION = "v1";
+    private static final String API_METER_ENDPOINT = "meters";
+    private static final String API_READINGS_ENDPOINT = "readings";
+
+    private final GsonBuilder gson_Builder = new GsonBuilder().registerTypeAdapter(ReadingInstance.class,
+            new CustomReadingInstanceDeserializer());
+    private final Gson gson = gson_Builder.create();
+
+    private String ressourceID;
+    private String meterID;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public MeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No handling needed, since we have only values to read.
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Pixometer handler '{}'", getThing().getUID());
+        updateStatus(ThingStatus.UNKNOWN);
+
+        try {
+
+            Configuration config = getThing().getConfiguration();
+            setRessourceID((String) config.get(CONFIG_THING_RESSOURCE_ID));
+
+            Bridge b = this.getBridge();
+            if (b == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "Could not find bridge (pixometer config). Did you choose one?");
+                return;
+            }
+
+            String token = new StringBuilder("Bearer ")
+                    .append(b.getConfiguration().get(CONFIG_BRIDGE_AUTH_TOKEN).toString()).toString();
+
+            obtainMeterId(token);
+
+            // Start polling job with the interval, that has been set up in the bridge
+            int pollingPeriod = Integer.parseInt(b.getConfiguration().get(CONFIG_BRIDGE_REFRESH).toString());
+            pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    logger.debug("Try to refresh meter data");
+                    try {
+                        updateMeter(token);
+                    } catch (RuntimeException r) {
+                        updateStatus(ThingStatus.OFFLINE);
+                        throw r;
```

Comment:
Re-throwing the error here will kill the scheduled job and only re-enable the binding when restarted (for example by editing and saving the configuration). So that may not something you want if it's a temporary failure.